app/src/main/java/com/epfl/drawyourpath/utils/Utils.kt

```diff
@@ -223,4 +228,32 @@ object Utils {
         val roundSpeed: Double = (speed * 100.0).roundToInt() / 100.0
         return roundSpeed.toString()
     }
+
+    /**
+     * Converts a list of LatLng to a stroke.
+     * @param coordinates The coordinates we want to convert
+     * @return A Stroke object representing the coordinates in planar space.
+     */
+    fun coordinatesToStroke(coordinates: List<LatLng>): Stroke {
+        val builder = Stroke.builder()
+
+        for (coordinate in coordinates) {
+            builder.addPoint(coordinateToPoint(coordinate))
+        }
+
+        return builder.build()
+    }
+
+    /**
+     * Converts a LatLng to a Point.
+     * @param coordinate The coordinate we want to convert
+     * @return A Point object representing the coordinate in planar space.
+     */
+    fun coordinateToPoint(coordinate: LatLng): Point {
+        val lat = Math.toRadians(coordinate.latitude)
+        val long = Math.toRadians(coordinate.longitude)
+        // Mercator projection formula
+        val y = ln(tan(lat) + (1 / cos(lat)))
+        return Point.create(long.toFloat(), y.toFloat())
```

Comment:
Neat formula for the Mercator projection !
There are many ways to project but it is probably the best one for this application, as this is the most common projection and it is not too expensive to compute.

Reply:
Yes, it is very cheap. But it applies some stretching around the poles. To see if it poses a problem for later.